# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelChild.cpp
# Commit: 398fc19f5f6a
# Full Hash: 398fc19f5f6adf0a58ddb43e2efbaac571db320b
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-24 09:31:07
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P13 Avoid ODA message being processed when the channel is already failed, r=mayhemer,kershaw,necko-reviewers
#   
#   One line fix with a bunch of explanation.
#   
#   When we perform networking on the socket process, OnDataAvailable could be sent directly from socket process to child process.
# ==============================================================================

diff -r 25822279ba5b -r 398fc19f5f6a netwerk/protocol/http/HttpChannelChild.cpp
--- a/netwerk/protocol/http/HttpChannelChild.cpp	Tue Jun 23 16:58:40 2020 +0000
+++ b/netwerk/protocol/http/HttpChannelChild.cpp	Tue Jun 23 17:02:55 2020 +0000
@@ -764,7 +764,7 @@
     return;
   }
 
-  if (mCanceled) {
+  if (mCanceled || NS_FAILED(mStatus)) {
     return;
   }
 
